src/main/java/model/rates/RateFactory.java

```diff
@@ -0,0 +1,28 @@
+package model.rates;
+
+class RateFactory {
+
+    Rate getRate(RateTypes type) {
+        Rate toReturn = null;
+        switch (type) {
+            case _4G:
+                toReturn = new Rate4G();
+                break;
+            case ULTRA:
+                toReturn = new RateUltra();
+                break;
+            case SUPER_8:
+                toReturn = new RateSuper8();
+                break;
+            case INTERNET_MIN:
+                toReturn = new RateInternetMin();
+                break;
+            case INTERNET_MAX:
+                toReturn = new RateInternetMax();
+                break;
+            default:
+                throw new IllegalArgumentException("Wrong rate type:" + type);
+        }
+        return toReturn;
```

Comment:
There is one approach in development called contract development. It means that every method guarantees that it will return valid results. My assumption here that null is not valid result. It forced every client of RateFactory to deal with this null return value.